src/components/NavBar/NavBar.vue

```diff
@@ -55,35 +70,75 @@ export default {
   name: "NavBar",
   data() {
     return {
-      stocks: [
-        { symbol: "AAPL", name: "Apple Inc." },
-        { symbol: "TSLA", name: "Tesla Inc." },
-        { symbol: "FSR", name: "Fisker Inc." },
-        { symbol: "SPI", name: "Spi Energy Co Ltd" },
-      ],
+      stocks: [],
       isLoading: false,
+      searchModel: null,
+      search: null,
+      searchQueue: [],
     };
   },
+  computed: {
+    fields() {
+      if (!this.searchModel) return [];
+      return Object.keys(this.searchModel).map((key) => {
+        return {
+          key,
+          value: this.searchModel[key] || "n/a",
+        };
+      });
+    },
+    items() {
+      return this.stocks.map((stock) => {
+        const Description = stock["1. symbol"] + ": " + stock["2. name"];
+        return Object.assign({}, stock, { Description });
+      });
+    },
+  },
+  watch: {
+    async search(val) {
+      // Checks if the user has typed anything in the last 2 seconds if not make a request to backend
+      this.searchQueue.push(val);
+      await new Promise((_) => setTimeout(_, 2000));
+      this.searchQueue.pop();
+      if (this.searchQueue.length == 0) {
+        if(val.length == 0){
+          return
+        }
+        if (this.isLoading) {
+          return;
+        }
+
+        this.isLoading = true;
+
+        // Lazily load input items
+        fetch(`http://localhost:5000/api/v1/searchRecommendations/${val}`)
```

Comment:
you will need to change the host as this wont work when it is deployed, would be better if u can use our own existing axios instance perform this API call. take a look at `axios/axios.v1.js` :)

Reply:
Right thanks I'll take a look

Reply:
Done